tools/idf.py

```diff
@@ -713,7 +713,32 @@ def main() -> None:
         else:
             raise
     else:
-        cli(sys.argv[1:], prog_name=PROG, complete_var=SHELL_COMPLETE_VAR)
+        argv = expand_file_arguments(argv or sys.argv[1:])
+
+        cli(argv, prog_name=PROG, complete_var=SHELL_COMPLETE_VAR)
+
+
+def expand_file_arguments(argv):
+    """
+    Any argument starting with "@" gets replaced with all values read from a text file.
+    Text file arguments can be split by newline or by space.
+    Values are added "as-is", as if they were specified in this order
+    on the command line.
+    """
+    new_args = []
+    expanded = False
+    for arg in argv:
+        if arg.startswith("@"):
+            expanded = True
+            with open(arg[1:], "r") as f:
+                for line in f.readlines():
+                    new_args += shlex.split(line)
+        else:
+            new_args.append(arg)
+    if expanded:
+        print("idf.py %s" % (" ".join(new_args[1:])))
+        return new_args
+    return argv
```

Comment:
`new_args` and `argv` should be the same even when there is no `@file.txt`
```suggestion
    return new_args
```